src/SignalR/common/Protocols.Json/src/Protocol/JsonHubProtocol.cs

```diff
@@ -187,7 +188,25 @@ public ReadOnlyMemory<byte> GetMessageBytes(HubMessage message)
                         }
                         else if (reader.ValueTextEquals(TargetPropertyNameBytes.EncodedUtf8Bytes))
                         {
+#if NETCOREAPP
+                            reader.Read();
+
+                            if (reader.TokenType != JsonTokenType.String)
+                            {
+                                throw new InvalidDataException($"Expected '{TargetPropertyName}' to be of type {JsonTokenType.String}.");
+                            }
+
+                            if (!reader.HasValueSequence)
+                            {
+                                target = binder.GetTarget(reader.ValueSpan) ?? Encoding.UTF8.GetString(reader.ValueSpan);
+                            }
+                            else
+                            {
+                                target = reader.GetString();
+                            }
+#else
                             target = reader.ReadAsString(TargetPropertyName);
+#endif
```

Comment:
Why do we need to cross-compile? Don't we already have `System.Memory` and `System.Buffers` dependency in netstandard2.0 for `ReadOnlySequence<byte>`?